src/Events/Controllers/CollectController.cs

```diff
@@ -77,7 +77,15 @@ public async Task<IActionResult> Post([FromBody] IEnumerable<EventModel> model)
                     }
                     if (cipher == null)
                     {
-                        continue;
+                        // When the user cannot access the cipher directly, check if the organization allows for
+                        // admin/owners access to all collections and the user can access the cipher from that perspective.
+                        var org = await _organizationRepository.GetByIdAsync(eventModel.OrganizationId.Value);
```

Comment:
🎨 We have a service `IApplicationCacheService` that provides a method called `GetOrganizationAbilityAsync()` that accepts and `organizationId`. It uses a cache to reduce the number of queries against the database for commonly used properties on organizations.

We should try and use that here instead of the `organizationRepoistory` if possible.

Reply:
⛏️ We should have a check in case the `organizationId` is missing, as is this looks like it will throw an exception trying to access the `.Value`.

Reply:
>We should have a check in case the organizationId is missing, as is this looks like it will throw an exception trying to access the .Value.

👍 [7fbd875](https://github.com/bitwarden/server/pull/4705/commits/7fbd875a6e01ba1d150df36275412174fc0d902e)